Firestore/Example/Tests/SpecTests/FSTSyncEngineTestDriver.h

```diff
@@ -294,12 +294,14 @@ typedef std::unordered_map<firebase::firestore::auth::User,
 @property(nonatomic, assign, readonly) const firebase::firestore::auth::User &currentUser;
 
 /** The set of active targets as observed on the watch stream. */
-@property(nonatomic, strong, readonly)
-    NSDictionary<FSTBoxedTargetID *, FSTQueryData *> *activeTargets;
+- (const std::unordered_map<firebase::firestore::model::TargetId, FSTQueryData *> &)activeTargets;
 
 /** The expected set of active targets, keyed by target ID. */
-@property(nonatomic, strong, readwrite)
-    NSDictionary<FSTBoxedTargetID *, FSTQueryData *> *expectedActiveTargets;
+- (const std::unordered_map<firebase::firestore::model::TargetId, FSTQueryData *> &)
```

Comment:
Very optional: Is it possible to just implement the getter/setter without actually changing from property to methods? It is equivalent but property is so dominant in ObjC coding.

Reply:
The issue here is specifically that Objective-C does not deal with constness in a reasonable way. If you declare a property the generated getter always returns the same type as the underlying value, leading to unnecessary copies. Conventionally in C++ we make the getters return a const reference which you just can't accomplish with Objective-C properties.